firebase-firestore/src/main/java/com/google/firebase/firestore/util/FirestoreChannel.java

```diff
@@ -77,14 +77,46 @@ public FirestoreChannel(
     FirestoreCallCredentials firestoreHeaders = new FirestoreCallCredentials(credentialsProvider);
     FirestoreStub firestoreStub =
         FirestoreGrpc.newStub(grpcChannel).withCallCredentials(firestoreHeaders);
-    this.channel = firestoreStub.getChannel();
+    this.channel = grpcChannel;
     this.callOptions = firestoreStub.getCallOptions();
 
     this.resourcePrefixValue =
         String.format(
             "projects/%s/databases/%s", databaseId.getProjectId(), databaseId.getDatabaseId());
   }
 
+  /**
+   * Shuts down the grpc channel. This is not reversible and renders the FirestoreChannel unusable.
+   */
+  public void shutdown() {
+    channel.shutdown();
+    try {
+      if (!channel.awaitTermination(100, TimeUnit.MILLISECONDS)) {
```

Comment:
Same comment about network RTT applies here. We're not in any terrible rush.

Reply:
Done.